Rangechecker.c

```diff
@@ -0,0 +1,62 @@
+#include "Rangechecker.h"
+
+int * sortinputarray(int *list, int size){
+    int i, j, temp;
+    for (i = 0; i < size - 1; ++i)
+    {
+        for (j = 0; j < size - 1 - i; ++j )
+        {
+            if (list[j] > list[j+1])
+            {
+                temp = list[j+1];
+                list[j+1] = list[j];
+                list[j] = temp;
+            }
+        }
+    }
+
+    return list;    
+}
+
+
+int checkrangesofsortedarray(range * rangelist, int *sortedlist, int size){
+     
+    int no_of_Ranges = 0;
+    rangelist[no_of_Ranges].startindex = sortedlist[0];
+    rangelist[no_of_Ranges].rangecount = 1;
+    for (int index = 0; index < size; index++)
+    {
+        if ((sortedlist[index] == sortedlist[index + 1]) || (sortedlist[index] + 1 == sortedlist[index + 1]))
```

Comment:
accessing the element sortedlist[index + 1] when index is of the last element, could lead to a garbage value or crash.
The design should handle an make sure it does not access beyond range element.
Please modify the design to prevent undefined behavior.